prov/sockets/src/sock_ctx.c

```diff
@@ -153,5 +153,6 @@ void sock_tx_ctx_abort(struct sock_tx_ctx *tx_ctx)
 {
 	rbfdabort(&tx_ctx->rbfd);
 	fastlock_release(&tx_ctx->rlock);
+	fastlock_release(&tx_ctx->wlock);
```

Comment:
I need help understanding this.  In this same file, I can see where sock_tx_ctx_start() acquires the wlock.  commit() releases the wlock.  abort() releases the rlock.  It makes sense that abort() should release the wlock.  But where was rlock ever acquired?  At the very least, the rlock should be acquired by some call inside this same source file, otherwise trying to track down where the locks were acquired and how is extremely difficult.

For example, why does abort release the rlock, but commit does not?